src/decaylanguage/dec/dec.py

```diff
@@ -277,7 +277,8 @@ def load_grammar(self, filename=None, parser="lalr", lexer="standard", **options
             # Conversion to handle pathlib on Python < 3.6:
             filename = str(filename)
 
-            self._grammar = open(filename).read()
+            with open(filename) as f:
+                self._grammar = f.read()
```

Comment:
This is a bug, the file was not being closed. Using `-Werror` causes this bug to show up.

Reply:
Indeed :S.

For my interest in which comment did you run -Werror? That's not something I've been playing with, it seems.

Reply:
I ran `nox -s tests -- -Werror`; if you are directly running pytest, you can run `pytest -Werror`. I considered making it default, or adding it to the GitHub Actions run, but that would turn all warnings into errors, and if there was a warning for something else we don't control, we might not want to break because of it.

Reply:
Nice! I should check more carefully all the pytest options. Thanks for the tip.

Reply:
Yes, better not make it default for the reason you mention. Hopefully warnings won't come up too often.

Reply:
`--strict` can't be used without generating a warning, as it's now `--strict-markers` (which is much clearer, it actually tells you what it's being strict about), but we can't use that until we drop Python 2, since the last PyTest to support Python 2 didn't have `--strict-markers` yet. That's one example (though I just removed it for now, so it technically does pass with `-Werror` at the moment).